src/probnum/statespace/sde.py

```diff
@@ -298,6 +353,45 @@ def f(t, y):
 
         return f, y0
 
+    def _setup_vectorized_mde_forward_sqrt(self, initrv, _diffusion=1.0):
+        """Set up forward moment differential equations (MDEs) using a square-root
+        implementation. (https://ieeexplore.ieee.org/document/4045974)
+
+        Compute an ODE vector field that represents the MDEs and is
+        compatible with scipy.solve_ivp.
+        """
+        dim = len(initrv.mean)
+
+        def f(t, y):
+            # Undo vectorization
+            mean, cov_sqrt_flat = y[:dim], y[dim:]
+            cov_sqrt = cov_sqrt_flat.reshape((dim, dim))
+
+            # Apply iteration
+            F = self.driftmatfun(t)
+            u = self.forcevecfun(t)
+            L = self.dispmatfun(t)
+
+            new_mean = F @ mean + u
+            cov_sqrt_lu, cov_sqrt_piv = scipy.linalg.lu_factor(cov_sqrt)
+            F_bar = scipy.linalg.lu_solve((cov_sqrt_lu, cov_sqrt_piv), F @ cov_sqrt)
+            L_bar = np.sqrt(_diffusion) * scipy.linalg.lu_solve(
+                (cov_sqrt_lu, cov_sqrt_piv), L
+            )
```

Comment:
isnt this sqrt a cholesky factor? if so, cho_solve would not only be faster but also more stable numerically

Reply:
I initially thought that we cannot rely on `cov_sqrt` being positive definite but you're right it is initialised using `initrv.cov_cholesky`.

Reply:
I just looked into this again realizing that I did not use a Cholesky decomposition since `cov_sqrt` is not symmetric. Sorry for the confusion!

Reply:
@pnkraemer here